File: golang/values.go
package abi

type U8Value struct {
	Value uint8
}

type U16Value struct {
	Value uint16
}

type U32Value struct {
	Value uint32
}

type U64Value struct {
	Value uint64
}

type I8Value struct {
	Value int8
}

type I16Value struct {
	Value int16
}

type I32Value struct {
	Value int32
}

type I64Value struct {
	Value int64
}

type BytesValue struct {
	Value []byte
}

type StringValue struct {
	Value string
}

type BoolValue struct {
	Value bool
}

type OptionValue struct {
	Value interface{}
}

type Field struct {
	Name  string
	Value interface{}
}

type StructValue struct {
	Fields []Field
}

type TupleValue struct {
	Fields []Field
}

type InputListValue struct {
	Items []interface{}
}

type OutputListValue struct {
	Items       []interface{}
	ItemCreator func() interface{}
}

type MultiValue struct {
	Items []interface{}
}

type OutputVariadicValues struct {
	Items       []interface{}
	ItemCreator func() interface{}
}

type OptionalValue struct {
	Value interface{}
	IsSet bool
}

File: python/multiversx_sdk_abi/values.py
from typing import Any, Callable, List


class U8Value:
    def __init__(self, value: int = 0):
        self.value = value


class U16Value:
    def __init__(self, value: int = 0):
        self.value = value


class U32Value:
    def __init__(self, value: int = 0):
        self.value = value


class U64Value:
    def __init__(self, value: int = 0):
        self.value = value


class I8Value:
    def __init__(self, value: int = 0):
        self.value = value


class I16Value:
    def __init__(self, value: int = 0):
        self.value = value


class I32Value:
    def __init__(self, value: int = 0):
        self.value = value


class I64Value:
    def __init__(self, value: int = 0):
        self.value = value


class BytesValue:
    def __init__(self, value: bytes = b''):
        self.value = value


class StringValue:
    def __init__(self, value: str = ''):
        self.value = value


class BoolValue:
    def __init__(self, value: bool = False):
        self.value = value


class OptionValue:
    def __init__(self, value: Any = None):
        self.value = value


class Field:
    def __init__(self, name: str = "", value: Any = None):
        self.name = name
        self.value = value


class StructValue:
    def __init__(self, fields: List[Field] = []):
        self.fields = fields


class TupleValue:
    def __init__(self, fields: List[Any] = []):
        self.fields = fields


class EnumValue:
    def __init__(self, discriminant: int = 0, fields: List[Field] = []):
        self.discriminant = discriminant
        self.fields = fields


class InputListValue:
    def __init__(self, items: List[Any] = []):
        self.values = items


class OutputListValue:
    def __init__(self, item_creator: Callable[[], Any]):
        self.items = []
        self.item_creator = item_creator


class InputMultiValue:
    def __init__(self, items: List[Any] = []):
        self.items = items


class InputVariadicValues:
    def __init__(self, items: List[Any] = []):
        self.items = items


class OutputMultiValue:
    def __init__(self, item_creator: Callable[[], Any]):
        self.items = []
        self.item_creator = item_creator


class OutputVariadicValues:
    def __init__(self, item_creator: Callable[[], Any]):
        self.items = []
        self.item_creator = item_creator


class OptionalValue:
    def __init__(self, value: Any = None):
        self.value = value
        self.is_set = value is not None
